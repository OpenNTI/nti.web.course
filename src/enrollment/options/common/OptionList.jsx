import './OptionList.scss';
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

export default class CourseEnrollmentOptionList extends React.Component {
	static propTypes = {
		className: PropTypes.string,
		options: PropTypes.arrayOf(
			PropTypes.shape({
				ListItem: PropTypes.func.isRequired
			})
		),
		selected: PropTypes.object,
		selectOption: PropTypes.func
	}


	selectOption = (option) => {
		const {selectOption} = this.props;

		if (selectOption) {
			selectOption(option);
		}
	}


	render () {
		const {options, selected, className} = this.props;

		return (
			<ul className={cx('nti-course-enrollment-option-list', className, {selected})}>
				{options.map((option, key) => {
					const {ListItem} = option;
					const isSelected = option === selected;

					return (
						<li key={key} className={cx({selected: isSelected && options.length > 1})}>
							<ListItem option={option} onSelect={this.selectOption} selected={isSelected}/>
						</li>
					);
				})}
			</ul>
		);
	}
}
