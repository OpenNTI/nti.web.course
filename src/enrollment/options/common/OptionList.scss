@import "~@nti/style-common";

.nti-course-enrollment-option-list {
	padding: 5px 1rem;
	list-style: none;
	margin: 0;

	li {
		margin: 5px 0;
		padding: 0.625rem 1rem;
		box-shadow: 0 0 0 1px var(--border-grey)-light;
		border-radius: 3px;

		&.selected {
			box-shadow: 0 0 0 2px var(--primary-blue);
		}
	}
}
