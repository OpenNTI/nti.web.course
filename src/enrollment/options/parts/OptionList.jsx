import React from 'react';
import PropTypes from 'prop-types';

export default class CourseEnrollmentOptionList extends React.Component {
	static propTypes = {
		options: PropTypes.arrayOf(
			PropTypes.shape({
				ListItem: PropTypes.func.isRequired
			})
		),
		selectedOption: PropTypes.object,
		selectOption: PropTypes.func
	}


	selectOption = (option) => {
		const {selectOption} = this.props;

		if (selectOption) {
			selectOption(option);
		}
	}


	render () {
		const {options, selectedOption} = this.props;

		return (
			<ul className="nti-course-enrollment-option-list">
				{options.map((option, key) => {
					const {ListItem} = option;

					return (
						<li key={key}>
							<ListItem option={option} selected={selectedOption === option} onSelect={this.selectOption} />
						</li>
					);
				})}
			</ul>
		);
	}
}
