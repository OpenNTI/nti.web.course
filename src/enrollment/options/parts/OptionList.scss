.nti-course-enrollment-option-list {
	padding: 1rem 0.625rem 0;
	list-style: none;
	margin: 0;
}
