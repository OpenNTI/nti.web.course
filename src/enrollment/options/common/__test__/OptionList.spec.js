import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';

import OptionList from '../OptionList';

class TestCmp extends React.Component {
	static propTypes = {
		option: PropTypes.object,
		selected: PropTypes.bool
	}

	render () {
		const {option, selected} = this.props;

		return (
			<div>
				<div>Option {option.id}</div>
				{selected && <div>Selected</div>}
			</div>
		);
	}
}

/* eslint-env jest */
describe('Course enrollment options common option list ', () => {
	test('Two items, second selected', async () => {
		const selected = {
			ListItem: TestCmp,
			id: '2'
		};

		const options = [
			{
				ListItem: TestCmp,
				id: '1'
			},
			selected
		];

		const cmp = renderer.create(<OptionList options={options} selected={selected}/>);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
