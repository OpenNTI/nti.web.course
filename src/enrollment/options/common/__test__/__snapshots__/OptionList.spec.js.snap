// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course enrollment options common option list  Two items, second selected 1`] = `
<ul
  className="nti-course-enrollment-option-list selected"
>
  <li
    className=""
  >
    <div>
      <div>
        Option 
        1
      </div>
    </div>
  </li>
  <li
    className="selected"
  >
    <div>
      <div>
        Option 
        2
      </div>
      <div>
        Selected
      </div>
    </div>
  </li>
</ul>
`;
